src/io/filewatcher.rs

```diff
@@ -53,17 +56,33 @@ impl FileWatcher {
         // Reset if file has been remade (i.e. is shorter) and update state
         if (meta.len() as usize) < self.last_size {
             start_idx = 0;
+            self.file = OpenOptions::new()
+                .read(true)
+                .write(false)
+                .open(&self.file_path)
+                .context("Failed to reopen file.")?;
         }
 
         self.last_size = meta.len() as usize;
-
+        
         // Get new file contents
-        let mut file =
-            FileWatcher::open_file(&self.file_path).context("Failed to reopen log file")?;
+        // self.file.seek(SeekFrom::Start(start_idx as u64))?;
         let mut buff: Vec<u8> = Vec::new();
-        let _ = file.read_to_end(&mut buff);
+        let read_size = self.file.read_to_end(&mut buff)?;
+
+        // If the length of the data we received is less than we expected, reopen the file and try again
+        if read_size < (self.last_size - start_idx) {
+            self.file = OpenOptions::new()
+                .read(true)
+                .write(false)
+                .open(&self.file_path)
+                .context("Failed to reopen file.")?;
+            // self.file.seek(SeekFrom::Start(start_idx as u64))?;
+            buff.clear();
+            let _ = self.file.read_to_end(&mut buff).context("Failed to reopen file.")?;
```

Comment:
Context here should be that it failed to read from the file rather than opening it. Although I notice the other read call doesn't have any context, so up to you if you remove this one or update it. If you update this one you should probably also add it to the other read call.